angularSTEP/src/app/map/map.component.ts

```diff
@@ -0,0 +1,167 @@
+import {
+  Component,
+  ViewChild,
+  ElementRef,
+  OnInit,
+  Input,
+  NgZone,
+} from '@angular/core';
+import {FormGroup, FormBuilder} from '@angular/forms';
+
+@Component({
+  selector: 'app-map',
+  templateUrl: './map.component.html',
+  styleUrls: ['./map.component.scss'],
+})
+export class MapComponent implements OnInit {
+  @Input() addressType!: string;
+  @ViewChild('addresstext') addresstext: any;
+
+  mapsForm: FormGroup;
+  autocompleteInput!: string;
+
+  @ViewChild('mapElem', {static: true}) mapElem!: ElementRef;
+  address = '';
+  center: google.maps.LatLngLiteral = {lat: 39.068229, lng: -94.393225};
```

Comment:
Consider starting with the user's current location:
https://medium.com/@balramchavan/display-and-track-users-current-location-using-google-map-geolocation-in-angular-5-c259ec801d58

Reply:
Thanks! This was a great idea and super easy to implement!